### PR TITLE
Optics.Re and documentation

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: optics-core/src/Optics.hs [hunk 1]
```diff
@@ -109,15 +109,83 @@ module Optics
...
+  -- 'x'
+  --
+  -- /Note:/ there are no @from@ combinator.
 
-  -- ** Re
+  , module Optics.Re
 
-  -- | TODO
---  , module Optics.Re
+  -- ** Indexed optics
```
Comment (adamgundry) [inline optics-core/src/Optics.hs]: I guess this section may need a bit of tweaking if we do #41.

File: optics-core/src/Optics.hs [hunk 2]
```diff
@@ -109,15 +109,83 @@ module Optics
...
 
-  -- ** Indexed optics
+  -- ** Re
 
-  -- | TODO
-  , module Optics.Indexed
+  -- | Some optics can be reversed with 're':
+  -- @'Iso' i s t a b@ into @'Iso' i b a t s@,
+  -- @'Getter' s t a b@ into @'Review' i b a t s@ etc.
+  -- Red arrows illustrate how 're' transforms optics:
```
Comment (adamgundry) [inline optics-core/src/Optics.hs]: This is nice, thanks. The "Core definitions" section above still mentions `re`, even though the diagram no longer includes the red arrows.

File: optics-core/src/Optics.hs (hunk 1 above)
Comment (phadej) [reply optics-core/src/Optics.hs]: True. I'll plan to decide (and implement) #41 first.

File: optics-core/src/Optics.hs [hunk 3]
```diff
@@ -109,15 +109,83 @@ module Optics
...
+  -- >>> let _Identity = iso runIdentity Identity
+  -- >>> view1 (_1 % re _Identity) ('x', "yz")
+  -- Identity 'x'
+  --
+  -- Yet we can use a 'Lens' as a 'Review' too:
+  --
+  -- >>> review (re _1) ('x', "yz")
+  -- 'x'
+  --
+  -- /Note:/ there are no @from@ combinator.
```
[the remark this answers is not in the document]
Comment (phadej) [reply optics-core/src/Optics.hs]: Made an issue to discuss that

File: optics-core/src/Optics.hs (hunk 1 above)
Comment (phadej) [reply optics-core/src/Optics.hs]: I made some corrections, but need to think properly how to approach indexed optics in the docs.

**Pull request merged**